src/main/scala/org/constellation/p2p/PeerAPI.scala

```diff
@@ -228,26 +230,28 @@ class PeerAPI(override val ipManager: IPManager, nodeActor: ActorRef)(
     createRoute(BlockBuildingRoundRoute.pathPrefix)(
       () => new BlockBuildingRoundRoute(nodeActor).createBlockBuildingRoundRoutes()
     )
-  private val mixedEndpoints = {
+
+  private[p2p] val mixedEndpoints = {
     path("transaction") {
       put {
-        entity(as[Transaction]) { tx =>
+        entity(as[TransactionGossip]) { gossip =>
+          logger.debug(s"Received transaction tx=${gossip.hash} with path=${gossip.path}")
           dao.metrics.incrementMetric("transactionRXByPeerAPI")
 
-          onComplete {
-            dao.transactionService
-              .contains(tx.hash)
-              .flatMap {
-                case false => dao.transactionService.put(TransactionCacheData(tx), as = TransactionStatus.Unknown)
-                case _     => IO.unit
-              }
-              // TODO: Respond with initial tx validation
-              .map(_ => StatusCodes.OK)
-              .unsafeToFuture()
-          } {
-            case Success(statusCode) => complete(statusCode)
-            case Failure(_)          => complete(StatusCodes.InternalServerError)
-          }
+          implicit val random: Random = scala.util.Random
```

Comment:
Is the validation performed now ? If not please do not remove that `TODO`

Reply:
There is a separate issue for that #287